yagb-core/src/emulator/ppu/ppu-cgb.ts

```diff
@@ -389,6 +405,101 @@ export class PpuCgb extends PpuBase {
         }
     };
 
+    private invalidRead: ReadHandler = (address) => {
+        this.system.log(`Invalid read from HDMA register ${hex16(address)}`);
+
+        return 0x00;
+    };
+
+    private hdma1Write: WriteHandler = (_, value) => {
+        this.hdmaSource = (this.hdmaSource & 0xff) | (value << 8);
+        this.hdmaSource &= 0xfff0;
+    };
+
+    private hdma2Write: WriteHandler = (_, value) => {
+        this.hdmaSource = (this.hdmaSource & 0xff00) | (value & 0xf0);
+        this.hdmaSource &= 0xfff0;
+    };
+
+    private hdma3Write: WriteHandler = (_, value) => {
+        this.hdmaDestination = (this.hdmaDestination & 0xff) | (value << 8);
+        this.hdmaDestination &= 0x1ff0;
+    };
+
+    private hdma4Write: WriteHandler = (_, value) => {
+        this.hdmaDestination = (this.hdmaDestination & 0xff00) | (value & 0xf0);
+        this.hdmaDestination &= 0x1ff0;
+    };
+
+    private hdma5Read: ReadHandler = () => {
+        if (this.hdmaLength === 0) return 0xff;
+
+        return (this.hdmaOn ? 0x00 : 0x80) | ((this.hdmaLength >> 4) - 1);
+    };
+
+    private hdma5Write: WriteHandler = (_, value) => {
+        if (this.hdmaOn && TransferMode.hblank === this.hdmaMode) {
```

Comment:
According to the cycle accurate gameboy doc, HDMA does not execute if the display is switched off. The correct behaviour is slightly is even more difficult, but I think returning if the LCD is off would be sufficient for now.